dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +66,24 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__isCertInstalled: function() {
+		var self = this, dispatchIfCertIsMissing = function(a, b) {
+			if (a + b !== 200 + 200) {
+				self.dispatcher.handleAppEvent({
+					name: "HTTPS_UNAVAILABLE",
+					status: xhr.status
+				});
+			}
+		};
+
+		self.client.ping("dashboard", "https").then(function (httpArgs) {
+			self.client.ping("controller", "https").then(function (httpsArgs) {
```

Comment:
that will make error detection much easier, `catch` only gets executed on network errors? Or on 404/500 as well?